File: 611. Valid Triangle Number/solution.java
class Solution {
    public int triangleNumber(int[] nums) {
        int n = nums.length;
        int rs = 0;
        if (n < 3) {
            return rs;
        }
        Arrays.sort(nums);
        for (int i = 0; i+2< n ; i++) {
            ok:for(int k= n-1 ; k>=i+2 ; k--){
                for(int j= i+1; j<k ; ++j){
                    if( nums[i] + nums[j] > nums[k] ){
                        rs += k- j ; 
                        continue ok ; 
                    }
                }
            }
            
        }

        return rs;
    }
}

File: 611. Valid Triangle Number/solution.py
class Solution:
    def triangleNumber(self, nums: List[int]) -> int:
        n : int = len( nums )
        nums.sort()
        rs = 0 
        for i in range( n-1 , 1 , -1 ) : 
            l = 0 
            r = i-1 
            while l<r : 
                if nums[l] + nums[r] > nums[i] : 
                    rs += r - l 
                    r -= 1 
                else : 
                    l += 1 
        return rs 